### DISCUSSION TITLE
[Chapter 03] 완전생성자 관련해서 궁금한 점

### DISCUSSION THREAD

**Discussion by snoopingh** [cowaine/is-my-code-that-strange#19] [Chapter 01-03 (설계)]
책 37p 관련해서

객체 초기화 시에 도메인 특성 상 일부 인스턴스 변수의 값이 없는 경우는 설계가 잘못 된 걸까요?
(초기에 의도적으로 null이 들어가서 본문 내용에 따르면 쓰레기 객체가 됨)

"값이없음"에 대한 코드값을 생성해서라도 반드시 초기화를 해야 하는 건지 궁금합니다.

**Comment by corock** [comment]
_e.g._ 임대라는 도메인이 있을 때, 인스턴스 초기화 시점에 부동산 매물에 대한 임대인이 아직 지정 안 된 경우?

**Comment by Irisation23** [comment]
회원등급 관리 (e.g BLANK, GREEN, BLUE)

``` java
public enum MemberGrade {
  BLANK, GREEN, BLUE
  
  // 기타 Overriding method 등등
}
```

으로 관리해 볼 수있음

**Comment by snoopingh** [comment]
자답) 필수로 초기화를 해야 하지만 그렇지 않은 경우에 대한 내용을 다루는 것으로 보임
질문자체는 책 내용을 곡해한 것으로...
